automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala

```diff
@@ -153,6 +160,104 @@ class SnapshotAPISpec extends AnyFreeSpecLike with Matchers with BeforeAndAfterA
+          createSnapshotReference(projectName, workspaceName, dataRepoSnapshotId, "testSnapshotForRunningAnalysis")
+
+          // validate the snapshot was added correctly: list snapshots in Rawls, should return 1, which we just added.
+          // if we can successfully list snapshot references, it means WSM created its copy of the workspace
+          val listResponse = listSnapshotReferences(projectName, workspaceName)
+
+          val resources = Rawls.parseResponseAs[DataReferenceList](listResponse).getResources.asScala
+          resources.size shouldBe 1
+          // TODO: update snapshot name?
+          resources.head.getName shouldBe "testSnapshotForRunningAnalysis"
+          resources.head.getReferenceType shouldBe ReferenceTypeEnum.DATA_REPO_SNAPSHOT
+          resources.head.getReference.getSnapshot shouldBe dataRepoSnapshotId
+
+          // create method config in a workspace
+          val createMethodConfigUrl  = Uri(Rawls.url).withPath(Path(s"/api/workspaces/$projectName/$workspaceName/methodconfigs"))
+          // TODO: a function createMethodConfigInWorkspace exists in workbench-libs, but not flexible/updated to accept dataReferenceName and maybe other parameters
+          // TODO: other ways to clean up payload building process in this test? move to a private method within this test class?
+          val createMethodConfigPayload = Map("methodRepoMethod" -> Map("methodUri" -> "agora://gatk/echo_to_file/9","methodName" -> "echo_to_file","methodNamespace" -> "gatk","methodVersion" -> 9),
+            "name" -> "echo_to_file-configured",
+            "namespace" -> "gatk",
+            "rootEntityType" -> "vcf_file",
+            "prerequisites" -> Map(),
+            "inputs" -> Map("echo_strings.echo_to_file.input1" -> "this.VCF_File_Name"),
+            "outputs" -> Map("echo_strings.echo_to_file.out" -> "workspace.output"),
+            "methodConfigVersion" -> 1,
+            "deleted" -> false,
+            "dataReferenceName" -> "testSnapshotForRunningAnalysis"
+          )
+          Rawls.postRequest(
+            uri = createMethodConfigUrl.toString(),
+            content = createMethodConfigPayload)
+
+          // run analysis on the snapshot
+          val createSubmissionUrl  = Uri(Rawls.url).withPath(Path(s"/api/workspaces/$projectName/$workspaceName/submissions"))
+          val createSubmissionPayload = Map(
+            "useCallCache" -> true,
+            "deleteIntermediateOutputFiles" -> false,
+            "methodConfigurationNamespace" -> "gatk",
+            "methodConfigurationName" -> "echo_to_file-configured"
+          )
+          val response = Rawls.postRequest(
+            uri = createSubmissionUrl.toString(),
+            content = createSubmissionPayload)
+          // TODO: revisit the mapper usage here
+          val submissionId = mapper.readTree(response).get("submissionId").asText()
```

Comment:
ideally, use the mapper to create a specific model class, then read the submission id from the model class. I think this is `SubmissionReport`

Reply:
Created AS-622 to keep track of this.